ipmasq: Add support for ip-masq-agent with IPv6 (#23219)
cilium-team/cilium: Documentation/network/concepts/masquerading.rst

From the output above, the program is running on the ``eth0`` and ``eth1`` devices.

> The eBPF-based masquerading can masquerade packets of the following L4 protocols:

Reviewer: Out of scope for this PR, but someday I'd love to see a clearer definition of masquerading in the page intro. We just sort of roll right into using it to mean a particular behavior without directly defining what it means when we use it. The current state isn't bad; I only think it could be better.

By default, all packets from a pod destined to an IP address outside of the
``ipv4-native-routing-cidr`` range are masqueraded, except for packets destined

> to other cluster nodes (similarly with ``ipv6-native-routing-cidr`` for IPv6).

Reviewer: obscure grammar nit, you could also go with "as with"
```suggestion
to other cluster nodes (similar to ``ipv6-native-routing-cidr`` for IPv6).
```

Author: Fixed, thank you

> The exclusion CIDR is shown in the above output of

Reviewer: Use active voice, s/above/preceding
```suggestion
The preceding output shows the exclusion CIDR of
```

Author: Fixed, thank you
